hack/verify-godep-licenses.sh

```diff
@@ -20,9 +20,15 @@ set -o pipefail
 
 KUBE_ROOT="$(cd "$(dirname "${BASH_SOURCE}")/.." && pwd -P)"
 
-branch="${1:-master}"
+readonly branch="origin/${1:-${KUBE_VERIFY_GIT_BRANCH:-master}}"
+echo "Checking for Godeps changes against ${branch}"
+# make sure the branch is valid, otherwise the next check will pass erroneously.
```

Comment:
Prefer to keep this branch stuff out of `verify-all.sh`, since not all checks use it.

I did move the logic to `hack/lib/util.sh` so at least it's not duplicated. (I also generalized it a bit so other verification checks could theoretically use it too.)